pyomo/contrib/pyros/separation_problem_methods.py

```diff
@@ -0,0 +1,482 @@
+            obj = Objective(expr=c.body - c.upper, sense=maximize)
+            map_obj_to_constr_names[c.name] = obj
+            model.add_component("separation_obj_" + str(idx), obj)
+            model.util.separation_objectives.append(obj)
+        elif c.lower is not None:
+            # This is an >= constraint, not supported
+            raise ValueError("All inequality constraints in model must be in standard form (<= RHS)")
+
+    model.util.map_obj_to_constr_names = map_obj_to_constr_names
+    for obj in model.util.separation_objectives:
+        obj.deactivate()
+
+    return
+
+
+def make_separation_problem(model_data, config):
+    """
+    Swap out uncertain param Param objects for Vars
+    Add uncertainty set constraints and separation objectives
+    """
+    separation_model = model_data.original.clone()
+
+    uncertain_params = separation_model.util.uncertain_params
+    separation_model.util.uncertain_param_vars = param_vars = Var(range(len(uncertain_params)))
+    map_new_constraint_list_names_to_original_con_names = {}
+
+    if config.objective_focus is ObjectiveType.worst_case:
+        separation_model.util.zeta = Param(initialize=0, mutable=True)
+        constr = Constraint(expr= separation_model.first_stage_objective + separation_model.second_stage_objective <= separation_model.util.zeta)
+        separation_model.add_component("epigraph_constr", constr)
+
+    substitution_map = {}
+    #Separation problem initialized to nominal uncertain parameter values
+    for idx, var in enumerate(list(param_vars.values())):
+        param = uncertain_params[idx]
+        var.value = param.value
+        substitution_map[id(param)] = var
+
+    separation_model.util.new_constraints = constraints = ConstraintList()
+
+    for c in separation_model.component_data_objects(Constraint):
+        num_uncertain_params_in_constr = len(list(i for i in list(identify_mutable_parameters(c.expr))
+                                                  if id(i) in list(id(u) for u in uncertain_params)))
+        if num_uncertain_params_in_constr > 0:
+            if c.equality:
+                constraints.add(
+                    replace_expressions(expr=c.lower, substitution_map=substitution_map) ==
+                    replace_expressions(expr=c.body, substitution_map=substitution_map))
+            elif c.lower is not None:
+                constraints.add(
+                    replace_expressions(expr=c.lower, substitution_map=substitution_map) <=
+                    replace_expressions(expr=c.body, substitution_map=substitution_map))
+            elif c.upper is not None:
+                constraints.add(
+                    replace_expressions(expr=c.upper, substitution_map=substitution_map) >=
+                    replace_expressions(expr=c.body, substitution_map=substitution_map))
+            c.deactivate()
+            map_new_constraint_list_names_to_original_con_names[list(constraints.values())[-1].name] = c.name
```

Comment:
I suppose I went with this approach because there are different clones of the model objects floating around and I wanted to be sure I was always getting the right components from the right model object. I will think about this more.